.github/actions/smoke_tests/action.yml

```diff
@@ -0,0 +1,34 @@
+name: 'smoke_tests'
+description: 'Executes smoke-tests'
+inputs:
+  relay-endpoint:
+    description: 'The endpoint of the relay e.g. relay.walletconnect.com'
+    required: false
+    default: 'relay.walletconnect.com'
+  project-id:
+    description: 'WalletConnect project id'
+    required: true
+  commit-sha:
+    description: 'Latest Commit in main SHA'
+    required: false
+
+runs:
+  using: "composite"
+  steps:
+    - uses: actions/cache/restore@v3
+      with:
+        path: |
+          products.tar
+        key: ${{ runner.os }}-deriveddata-main-${{ inputs.commit-sha }}
```

Comment:
Do we need to restore `SourcePackagesCache` for smoke_tests?

Reply:
In case of cache hit we should have DerivedData containing all products necessary to just execute tests with `test-without-building` so we don't need to `SourcePackagesCache`